tests/instruments/keysight/test_keysightN7744C.py

```diff
@@ -0,0 +1,39 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2025 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import pytest
+from pymeasure.test import expected_protocol
+from pymeasure.instruments.keysight.keysightN7744C import KeysightN7744C
+
+
+@pytest.mark.parametrize("channel", [1, 2, 3, 4])
+def test_wavelength(channel):
+    """Verify the wavelength setpoint setter and getter."""
+    with expected_protocol(
+        KeysightN7744C,
+        [(f"SENSe{channel}:POWer:WAVelength 1550NM", None),
+         (f"SENSe{channel}:POWer:WAVelength?", "1.55e-6")],
+    ) as inst:
+        inst.channels[channel].wavelength = 1550
+        assert inst.channels[channel].wavelength == pytest.approx(1550.0)
```

Comment:
You could separate the tests for getter and setter to better point to a possible error.

These protocol tests do not require a certain setting, i.e. the getter works without a specific setter beforehand.